packages/@lwc/babel-plugin-component/src/__tests__/utils/test-transform.js

```diff
@@ -5,86 +5,20 @@
 const BASE_CONFIG = { babelrc: false, configFile: false, filename: 'test.js' };
 
-function transform(plugin, pluginOpts = {}, opts = {}) {
-    const testConfig = Object.assign(
-        {},
-        BASE_CONFIG,
-        {
-            plugins: [[plugin, pluginOpts]],
-        },
-        opts
-    );
-
-    return function (source) {
-        return babel.transform(prettify(source), testConfig);
+module.exports = function transform(source, opts = {}) {
+    const testConfig = {
+        ...BASE_CONFIG,
+        plugins: [[plugin, opts]],
     };
-}
-
-function prettify(str) {
-    return str
-        .toString()
-        .replace(/^\s+|\s+$/, '')
-        .split('\n')
-        .map((line) => line.trim())
-        .filter((line) => line.length)
-        .join('\n');
-}
-
-function stripComments(str) {
-    return str.replace(/\/\*LWC compiler v[\d.]+\*\//g, '');
-}
-
-function pluginTest(plugin, pluginOpts, opts = {}) {
-    const testTransform = transform(plugin, pluginOpts, opts);
-
-    const transformTest = function (actual, expected) {
-        if (expected.error) {
-            let transformError;
 
-            try {
-                testTransform(actual);
-            } catch (error) {
-                transformError = error;
-            }
-
-            expect(transformError.toString()).toContain(expected.error.message);
-            expect(transformError.loc).toEqual(expected.error.loc);
-        } else if (expected.output) {
-            const output = testTransform(actual);
-            if (expected.output.code !== undefined) {
-                const normalizedActual =
-                    output && output.code && stripIndents(stripComments(output.code));
-                const normalizedExpected = stripIndents(expected.output.code);
-
-                if (normalizedActual !== normalizedExpected) {
-                    // we should fail, but with style
-                    expect(prettier.format(normalizedActual, { parser: 'babel' })).toBe(
-                        prettier.format(normalizedExpected, { parser: 'babel' })
-                    );
-                } else {
-                    expect(normalizedActual).toBe(normalizedExpected);
-                }
-            }
-        } else {
-            throw new TypeError(`Transform test expect an object with either error or output.`);
-        }
-    };
-
-    const pluginTester = (name, actual, expected) =>
-        test(name, () => transformTest(actual, expected));
-    pluginTester.skip = (name) => test.skip(name);
-    pluginTester.only = (name, actual, expected) => {
-        // eslint-disable-next-line jest/no-focused-tests
-        test.only(name, () => transformTest(actual, expected));
-    };
+    let { code } = babel.transform(source, testConfig);
 
-    return pluginTester;
-}
+    // Replace LWC's version with X.X.X so the snapshots don't frequently change
+    code = code.replace(new RegExp(LWC_VERSION.replace(/\./g, '\\.'), 'g'), 'X.X.X');
```

Comment:
We could run prettier on the Babel-generated output, but I'm not sure it's worth the extra overhead. If Babel's output formatting changes frequently, then maybe we would benefit from using something more stable like Prettier. For now, I kept Prettier out of here.